raco-sloc.rkt

```diff
@@ -0,0 +1,49 @@
+#lang racket/base
+
+;; Command-line interface for computing SLOC
+
+(require syntax-sloc
+         (only-in syntax-sloc/read-lang-file lang-file?)
+         racket/format)
+
+;; -----------------------------------------------------------------------------
+
+(define SLOC-HEADER "SLOC\tSource")
+
+(define MAX-PATH-WIDTH 40) ;; characters
+
+;; Get SLOC for `src`, output a string with pretty-printed results
+;; format-sloc : (Path-String -> Natural) Path-String -> String
+(define (format-sloc get-sloc src)
+  (string-append (~r (get-sloc src) #:min-width 4)
+                 "\t"
+                 (format-filepath src)))
+
+;; Print a filepath, truncate if too long
+;; format-filepath : Path-String -> String
+(define (format-filepath src)
+  (~a src #:max-width MAX-PATH-WIDTH
+          #:limit-marker "..."
+          #:limit-prefix? #t))
+
+;; missing-sloc : Path-String -> String
+(define (missing-sloc src)
+  (string-append " N/A" "\t" src))
```

Comment:
(I guess I should have looked at this a bit more closely) This is a type error. You can't apply `string-append` to a `Path-String`, because `string-append` can't take paths.